src/zui/ZUIButton/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { Button, CircularProgress } from '@mui/material';
+import { FC, KeyboardEventHandler, MouseEventHandler } from 'react';
+
+type ZUIButtonType =
+  | 'primary'
+  | 'secondary'
+  | 'tertiary'
+  | 'destructive'
+  | 'warning'
+  | 'loading';
+
+interface ZUIButtonProps {
+  buttonType?: 'button' | 'reset' | 'submit';
+  disabled?: boolean;
+  endIcon?: JSX.Element;
+  fullWidth?: boolean;
+  label: string | JSX.Element;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+  onKeyDown?: KeyboardEventHandler<HTMLButtonElement>;
+  size?: 'large' | 'medium' | 'small';
+  startIcon?: JSX.Element;
+  type?: ZUIButtonType;
```

Comment:
I'm wondering if a better name for this is "level" or "action type" or something like that? Or even "variant"?